perfkitbenchmarker/benchmarks/cloudsuite_web_serving.py

```diff
@@ -0,0 +1,411 @@
+                     'chmod a+rwx %s && '
+                     'chmod +x %s/benchmarks/OlioDriver/bin/fileloader.sh'
+                     % (APP_DIR, BASE_DIR, filestore, filestore,
+                        FABAN_HOME))
+    vm.RemoteCommand('export FILESTORE=%s && '
+                     'export JAVA_HOME=%s && '
+                     '%s/benchmarks/OlioDriver/bin/fileloader.sh 102 %s'
+                     % (filestore, JAVA_HOME, FABAN_HOME, filestore))
+    frontend.RemoteCommand('perl -pi -e '
+                           '"s/%s/%s/g"'
+                           ' %s/etc/config.php'
+                           % ('(\$olioconfig\[\'localfsRoot\'\]).*',
+                              '\$1 = \'%s\';' % re.escape(filestore),
+                              APP_DIR))
+    return
+
+  setupFilestore(frontend)
+  frontend.RemoteCommand('sudo cp /usr/local/etc/php-fpm.conf.default '
+                         '/usr/local/etc/php-fpm.conf && '
+                         'sudo /usr/local/sbin/php-fpm')
+  nginx.Start(frontend, benchmark_spec.firewall)
+  return
+
+
+def setupBackend(benchmark_spec):
+  vms = benchmark_spec.vms
+  backend = vms[BK]
+  frontend = vms[FR]
+  CLIENT_IP = vms[CL].ip_address
+  FRONTEND_IP = frontend.ip_address
+  backend.Install('libaio')
+  untar_command = ('cd %s && tar xzf %s')
+  backend.RemoteCommand(untar_command %
+                        (BASE_DIR, 'mysql-5.5.20-linux2.6-x86_64.tar.gz'))
+  copy_command = ('cd %s && sudo cp support-files/my-medium.cnf %s')
+  backend.RemoteCommand(copy_command % (MYSQL_HOME, MY_CNF))
+  db_install_command = ('cd %s && scripts/mysql_install_db')
+  backend.RemoteCommand(db_install_command % (MYSQL_HOME))
+  backend.RobustRemoteCommand('cd %s && ./bin/mysqld_safe &' % MYSQL_HOME)
+  time.sleep(15)
+  sql_cmd = 'create user \'olio\'@\'%\' identified by \'olio\';'
+  backend.RemoteCommand('cd %s && '
+                        './bin/mysql -uroot -e "%s"' % (MYSQL_HOME, sql_cmd))
+  backend.RemoteCommand('cd %s && ./bin/mysql -uroot -e "grant all privileges '
+                        'on *.* to \'olio\'@\'localhost\' '
+                        'identified by \'olio\' with grant option; '
+                        'grant all privileges on *.* to '
+                        '\'olio\'@\'%s\' identified by \'olio\' with grant '
+                        'option;"' % (MYSQL_HOME, FRONTEND_IP))
+  backend.RemoteCommand('cd %s && ./bin/mysql -uroot -e "create database olio;'
+                        'use olio; \. %s/benchmarks/OlioDriver/bin/schema.sql"'
+                        % (MYSQL_HOME, FABAN_HOME))
+  populate_db_command = ('export JAVA_HOME=%s && '
+                         'cd %s/benchmarks/OlioDriver/bin'
+                         '&& chmod +x dbloader.sh && ./dbloader.sh localhost '
+                         '%s')
+  backend.RobustRemoteCommand(populate_db_command
+                              % (JAVA_HOME, FABAN_HOME, LOAD_SCALE))
+  time.sleep(100)
```

Comment:
I notice these sleep statements throughout the setup with seemingly arbitrary times. What are they for?